Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/storage/internal/migrations/ContactDiaryDatabaseMigration2To3.kt

```diff
@@ -0,0 +1,45 @@
+package de.rki.coronawarnapp.contactdiary.storage.internal.migrations
+
+import androidx.room.migration.Migration
+import androidx.sqlite.db.SupportSQLiteDatabase
+import de.rki.coronawarnapp.exception.ExceptionCategory
+import de.rki.coronawarnapp.exception.reporting.report
+import timber.log.Timber
+
+/**
+ * Migrates the contact diary database from schema version 2 to schema version 3.
+ * We are adding additional columns for new optional attributes
+ * Location: TraceLocationGUID
+ * LocationVisit: CheckInID
+ */
+object ContactDiaryDatabaseMigration2To3 : Migration(2, 3) {
+
+    override fun migrate(database: SupportSQLiteDatabase) {
+        try {
+            Timber.i("Attempting migration 2->3...")
+            performMigration(database)
+            Timber.i("Migration 2->3 successful.")
+        } catch (e: Exception) {
+            Timber.e(e, "Migration 2->3 failed")
+            e.report(ExceptionCategory.INTERNAL, "ContactDiary database migration failed.")
+            throw e
```

Comment:
When an exception thrown here , what should should happen? is the database in a state where it can be used?